filecoindot-nft/src/lib.rs

```diff
@@ -0,0 +1,144 @@
+// Copyright 2021 ChainSafe Systems
+// SPDX-License-Identifier: LGPL-3.0-only
+
+//! # Filecoin NFT Pallet
+//!
+//! This pallet will enabled users to mint NFTs
+//! providing the cid and proof.
+//!
+#![cfg_attr(not(feature = "std"), no_std)]
+
+pub use self::pallet::*;
+
+mod types;
+
+#[frame_support::pallet]
+pub mod pallet {
+    pub use crate::types::{ClassData, TokenData};
+    use frame_support::{pallet_prelude::*, sp_std::prelude::*};
+    use frame_system::pallet_prelude::*;
+
+    pub type TokenIdOf<T> = <T as orml_nft::Config>::TokenId;
+    pub type ClassIdOf<T> = <T as orml_nft::Config>::ClassId;
+
+    const MINT_QUANTITY: u32 = 1;
+
+    /// Configure the pallet by specifying the parameters and types on which it depends.
+    #[pallet::config]
+    pub trait Config:
+        frame_system::Config + orml_nft::Config<ClassData = ClassData, TokenData = TokenData>
+    {
+        /// The overarching event type.
+        type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+        /// The default class id of the NFT
+        type DefaultClassId: Get<ClassIdOf<Self>>;
+        /// The weight for this pallet's extrinsics.
+        type WeightInfo: WeightInfo;
+    }
+
+    #[pallet::pallet]
+    #[pallet::generate_store(pub (super) trait Store)]
+    pub struct Pallet<T>(_);
+
+    #[pallet::event]
+    #[pallet::generate_deposit(pub(crate) fn deposit_event)]
+    pub enum Event<T: Config> {
+        /// Minted NFT token. \[from, class_id, quantity\]
+        MintedToken(T::AccountId, ClassIdOf<T>, u32),
+        /// Transferred NFT token. \[from, to, class_id, token_id\]
+        TransferredToken(T::AccountId, T::AccountId, ClassIdOf<T>, TokenIdOf<T>),
+    }
+
+    // Errors inform users that something went wrong.
+    #[pallet::error]
+    pub enum Error<T> {
+        /// The requested token id does not exist
+        TokenIdNotFound,
+    }
+
+    #[pallet::hooks]
+    impl<T: Config> Hooks<T::BlockNumber> for Pallet<T> {}
+
+    #[pallet::genesis_config]
+    pub struct GenesisConfig<T: Config> {
+        pub default_class: (T::AccountId, Vec<u8>),
+    }
+
+    #[cfg(feature = "std")]
+    impl<T: Config> Default for GenesisConfig<T> {
+        fn default() -> Self {
+            Self {
+                default_class: (Default::default(), Default::default()),
+            }
+        }
+    }
+
+    #[pallet::genesis_build]
+    impl<T: Config> GenesisBuild<T> for GenesisConfig<T> {
+        fn build(&self) {
+            let (owner, data) = self.default_class.clone();
+            // just panic if cannot create class
+            orml_nft::Pallet::<T>::create_class(&owner, vec![], ClassData::new(data))
+                .expect("cannot create default nft class");
+        }
+    }
+
+    #[pallet::call]
+    impl<T: Config> Pallet<T> {
+        // **************** Relayer Add/Remove *****************
+        /// Adds a new relayer to the relayer set.
```

Comment:
this document is for https://github.com/ChainSafe/filecoindot/blob/a5f96f7b79b41c3c71a3711da136e5413095baa7/filecoindot/src/lib.rs#L246